src/modules/interaction/ParticipantInteraction.tsx

```diff
@@ -1,147 +1,167 @@
-import { ReactElement, useEffect, useState } from 'react';
+import {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
 
 import { Button } from '@mui/material';
 import Box from '@mui/material/Box';
 import Typography from '@mui/material/Typography';
 
+import { useLocalContext } from '@graasp/apps-query-client';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { hooks, mutations } from '@/config/queryClient';
 import { START_INTERACTION_BUTTON_CY } from '@/config/selectors';
 import MessagesPane from '@/modules/message/MessagesPane';
 import Agent from '@/types/Agent';
 import AgentType from '@/types/AgentType';
+import Exchange from '@/types/Exchange';
 import Interaction from '@/types/Interaction';
 
+import { defaultSettingsValues, useSettings } from '../context/SettingsContext';
+
+// Main component: ParticipantInteraction
 const ParticipantInteraction = (): ReactElement => {
-  const participantId = '0';
+  // Getting the participant ID from local context
+  const { memberId: participantId } = useLocalContext();
 
-  const artificialAssistant: Agent = {
-    id: '1',
-    name: 'Interviewer',
-    description: 'Assistant Description',
+  const { data: appDatas } = hooks.useAppData<Interaction>();
+  const { mutate: postAppData } = mutations.usePostAppData();
+  const { mutate: patchAppData } = mutations.usePatchAppData();
+  const { chat, exchanges } = useSettings();
+
+  const { t } = useTranslation();
+
+  // Define a default user as an agent
+  const defaultUser: Agent = {
+    id: uuidv4(),
+    name: 'Default User',
+    description: 'Default user description',
+    type: AgentType.User,
+  };
+
+  // Define the current member as an agent, merging with the default user
+  const currentMember: Agent = {
+    ...defaultUser,
+    ...hooks
+      .useAppContext()
+      // Find the member in app context data by participant ID
+      .data?.members.find((member) => member.id === participantId),
+  };
+
+  // Define a default assistant as an agent
```

Comment:
Maybe these defaults should be factored out to a `config.ts` file?

Reply:
Added:
defaultUser
defaultAssistant
defaultExchange
defautlInteraction

to new
/confic/config.ts